zne/extrapolation/exponential_extrapolator.py

```diff
@@ -0,0 +1,165 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022-2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Polynomial extrapolator."""
+
+from __future__ import annotations
+
+from numpy import array, exp, inf, ndarray, ones, sqrt
+from scipy.optimize import curve_fit
+
+from zne.utils.grouping import group_elements_gen
+
+from .extrapolator import OLSExtrapolator, ReckoningResult
+
+
+################################################################################
+## GENERAL
+################################################################################
+class MultiExponentialExtrapolator(OLSExtrapolator):
+    """Multi-exponential ordinary-least-squares (OLS) extrapolator.
+
+    Theoretical results [1] point at the multi-exponential model being the
```

Comment:
This could be moved to a doc, and the docstring could be more software-centric. 

I haven't looked at your docs setup yet. It may make sense for this to be here if you don't have an appropriate doc for it to go in.

Reply:
I am trying to do docs within the code, but happy to hear any suggestions 🙂